src/components/calcite-modal/calcite-modal.tsx

```diff
@@ -100,8 +101,19 @@ export class CalciteModal {
     }
   }
 
+  componentDidLoad(): void {
+    this.observer?.observe(this.el, { childList: true, subtree: true });
+  }
+
+  connectedCallback(): void {
+    if (Build.isBrowser) {
```

Comment:
nitpick: Just curious, why the build check? Otherwise, you could just create the observer when you create the private property.

```
private observer: MutationObserver = new MutationObserver(this.updateFooterVisibility);
```

Reply:
honestly I copied this pattern from elsewhere and assumed there was good reason for it in terms of some hydrated error but if not I would love to get rid of it.

Reply:
Yeah i'm not sure when/why it started. My guess is that it was causing an error running in tests?

Reply:
@jcfranco do you know? Someone must know :)

Reply:
I see this: https://github.com/Esri/calcite-components/blob/master/conventions/README.md#prerendering-and-ssr

But it was from a couple years ago. I'm sure we have many places we're using browser APIs and not checking for the build flag.

Reply:
It's needed for server-side-rendering: https://stenciljs.com/docs/static-site-generation-basics#things-to-watch-for. I know it was a main focus for components in the beginning, but not sure how many people are using it in that way.

> I'm sure we have many places we're using browser APIs and not checking for the build flag.

True. I think anything in the index page will go through prerendering, so that could be a way to catch problems.

Reply:
Is there a way we can test for this? I know there are many places where we are setting up observers in componentConnected and we don't check the build flag. What should we do?